// ad-cypress-lib/cypress/support/contracts/CreateCompanyV2Contract.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _en_US = _interopRequireDefault(require("faker/locale/en_US"));

var _CompanySizeType = _interopRequireDefault(require("./CompanySizeType"));

var _CompanyAddressContract = _interopRequireDefault(require("./CompanyAddressContract"));

var _UserCompanyMembershipAccountV2Contract = _interopRequireDefault(require("./UserCompanyMembershipAccountV2Contract"));

var _DefaultContract2 = _interopRequireDefault(require("./DefaultContract"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var CreateCompanyV2Contract = /*#__PURE__*/function (_DefaultContract) {
  _inherits(CreateCompanyV2Contract, _DefaultContract);

  var _super = _createSuper(CreateCompanyV2Contract);

  // Note: using same class for both Marketplace CreateCompanyV2Request
  // and CreateCompanyV2Response.  May split later.
  function CreateCompanyV2Contract() {
    var _this;

    _classCallCheck(this, CreateCompanyV2Contract);

    // ref https://help.appdirect.com/api/appdirect.html?shell#createcompanyv2request
    _this = _super.call(this);
    _this.allowLogin = null; // Boolean, optional

    _this.customAttributes = null; // array of CustomAttributeAccountV2

    _this.defaultRole = null; // String

    _this.firstUser = null; // UserCompanyMembershipAccountV2

    _this.name = null; // String

    _this.phoneNumber = null; // String

    _this.picture = null; // String

    _this.reseller = null; // Boolean

    _this.vendor = null; // Boolean

    _this.address = null; // CompanyAddress

    _this.emailAddress = null; // String

    _this.companySize = null; // CompanySizeType

    _this.countryCode = null; // String - ISO ALPHA-2 standard
    // Response:

    _this.uuid = null; // String - uuid

    _this.enabled = null; // Boolean;

    return _this;
  }

  _createClass(CreateCompanyV2Contract, [{
    key: "fakeTestData",
    value: function fakeTestData(options) {
      options = options || {};
      this.address = options.address || new _CompanyAddressContract["default"]().fakeTestData();
      this.allowLogin = options.allowLogin || 'true';
      this.customAttributes = options.customAttributes || [];
      this.defaultRole = options.defaultRole || "USER";
      this.name = options.name || 'Cypress ' + _en_US["default"].company.companyName();
      this.phoneNumber = _en_US["default"].phone.phoneNumber('###-###-####');
      this.picture = options.picture || _en_US["default"].image.business() + '.jpg';
      this.reseller = options.reseller || true;
      this.vendor = options.vendor || true;
      this.emailAddress = options.emailAddress || 'cypress.test.' + _en_US["default"].random.number({
        min: 100000,
        max: 999999
      }) + '@' + _en_US["default"].internet.domainName();
      this.companySize = _CompanySizeType["default"].random();
      this.countryCode = 'US';
      this.firstUser = options.firstUser || new _UserCompanyMembershipAccountV2Contract["default"]().fakeTestData(this);
      return this;
    }
  }, {
    key: "get",
    value: function get() {
      return {
        allowLogin: this.allowLogin,
        customAttributes: this.customAttributes,
        defaultRole: this.defaultRole,
        firstUser: this.firstUser.get(),
        name: this.name,
        phoneNumber: this.phoneNumber,
        picture: this.picture,
        reseller: this.reseller,
        vendor: this.vendor,
        address: this.address !== null ? this.address.get() : null,
        emailAddress: this.emailAddress,
        countryCode: this.countryCode,
        enabled: this.enabled,
        companySize: this.companySize
      };
    }
  }, {
    key: "update",
    value: function update(options) {
      this.address = options.address !== null ? new _CompanyAddressContract["default"]().update(options.address) : null;
      this.allowLogin = options.allowLogin;
      this.customAttributes = options.customAttributes;
      this.defaultRole = options.defaultRole;
      this.firstUser = new _UserCompanyMembershipAccountV2Contract["default"]().update(options.firstUser);
      this.name = options.name;
      this.phoneNumber = options.phoneNumber;
      this.picture = options.picture;
      this.reseller = options.reseller;
      this.vendor = options.vendor;
      this.uuid = options.uuid;
      this.enabled = options.enabled;
      this.emailAddress = options.emailAddress;
      this.companySize = options.companySize;
      this.countryCode = options.countryCode;
      return this;
    }
  }]);

  return CreateCompanyV2Contract;
}(_DefaultContract2["default"]);

exports["default"] = CreateCompanyV2Contract;
;
