// ad-cypress-lib/cypress/support/contracts/UserMembershipCompanyV1Contract.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _DefaultContract2 = _interopRequireDefault(require("./DefaultContract"));

var _CreateCompanyContactV1Contract = _interopRequireDefault(require("./CreateCompanyContactV1Contract"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var UserMembershipCompanyV1Contract = /*#__PURE__*/function (_DefaultContract) {
  _inherits(UserMembershipCompanyV1Contract, _DefaultContract);

  var _super = _createSuper(UserMembershipCompanyV1Contract);

  function UserMembershipCompanyV1Contract() {
    var _this;

    _classCallCheck(this, UserMembershipCompanyV1Contract);

    _this = _super.call(this);
    _this.id = null; // String (UUID)

    _this.name = null; // String

    _this.enabled = null; // Boolean

    _this.contact = null; // CreateCompanyContactV1Contract

    _this.size = null; // String (CompanySizeType)

    _this.status = null; // String [ACTIVE, INACTIVE]

    _this.attributes = null; // TBD

    _this.creationDate = null; // Number

    _this.industry = null; // TBD

    _this.salesAgent = null; // TBD

    _this.website = null; // String

    _this.emailAddress = null; // String

    _this.dealer = null; // TBD

    _this.uuid = null; // String (UUID)

    _this.externalId = null; // String (UUID)

    _this.countryCode = null; // String

    _this.permissions = null; // Array

    _this.defaultIdpUuid = null; // String (UUID)

    _this.domains = null; // String

    return _this;
  }

  _createClass(UserMembershipCompanyV1Contract, [{
    key: "get",
    value: function get() {
      return {
        id: this.id,
        name: this.name,
        enabled: this.enabled,
        contact: this.contact !== null ? this.contact.get() : null,
        size: this.size,
        status: this.status,
        attributes: this.attributes,
        creationDate: this.creationDate,
        industry: this.industry,
        salesAgent: this.salesAgent,
        website: this.website,
        emailAddress: this.emailAddress,
        dealer: this.dealer,
        uuid: this.uuid,
        externalId: this.externalId,
        countryCode: this.countryCode,
        permissions: this.permissions,
        defaultIdpUuid: this.defaultIdpUuid,
        domains: this.domains
      };
    }
  }, {
    key: "update",
    value: function update(options) {
      // the update() method will also verify the response body fileds or contract.
      this.id = options.id;
      this.name = options.name;
      this.enabled = options.enabled;
      this.contact = options.contact !== null ? new _CreateCompanyContactV1Contract["default"]().update(options.contact) : null;
      this.size = options.size;
      this.status = options.status;
      this.attributes = options.attributes;
      this.creationDate = options.creationDate;
      this.industry = options.industry;
      this.salesAgent = options.salesAgent;
      this.website = options.website;
      this.emailAddress = options.emailAddress;
      this.dealer = options.dealer;
      this.uuid = options.uuid;
      this.externalId = options.externalId;
      this.countryCode = options.countryCode;
      this.permissions = options.permissions;
      this.defaultIdpUuid = options.defaultIdpUuid;
      this.domains = options.domains;
      return this;
    }
  }]);

  return UserMembershipCompanyV1Contract;
}(_DefaultContract2["default"]);

exports["default"] = UserMembershipCompanyV1Contract;
;
